ui/plugins/auto_classification/controller.js

```diff
@@ -1,372 +1,924 @@
 "use strict";
 
-treeherder.controller('ClassificationPluginCtrl', [
-    '$scope', '$rootScope', 'ThLog', 'ThFailureLinesModel', 'ThClassifiedFailuresModel',
-    'ThMatcherModel', '$q', 'thEvents', 'thTabs', '$timeout', 'thNotify',
-    function ClassificationPluginCtrl(
-        $scope, $rootScope, ThLog, ThFailureLinesModel, ThClassifiedFailuresModel,
-        ThMatcherModel, $q, thEvents, thTabs, $timeout, thNotify) {
-        var $log = new ThLog(this.constructor.name);
+// TODO: move these into services or something
```

Comment:
In fact, I'd consider it a personal favor if you'd split it up.  :)  At this point, I can't tell what changed with the ``ClassificationPluginCtrl``.  The diff just looks like a meaningless green and red striped candy cane.  And no one wants candy canes to be meaningless.  Christmas is not about code reviews.  We can all agree that Christmas is about one, unifying, theme: presents.